### PR TITLE
feat(bash): improve highlights

### PR DESCRIPTION
**Pull request by ObserverOfTime**
Waiting for tree-sitter/tree-sitter-bash#193 to land.

### CODE REVIEW

File: queries/bash/highlights.scm [hunk 1]
```diff
@@ -89,17 +81,23 @@
...
 (comment) @comment @spell
-(test_operator) @string
 
-(command_substitution
-  [ "$(" ")" ] @punctuation.bracket)
+(test_operator) @operator
+
+(command_substitution "$(" @punctuation.bracket)
+(process_substitution "<(" @punctuation.bracket)
+(arithmetic_expansion "$((" @punctuation.bracket)
```
Comment (amaanq) [inline queries/bash/highlights.scm:92]: hm, not special?

File: queries/bash/highlights.scm [hunk 2]
```diff
@@ -1,29 +1,15 @@
...
 [
  ";"
  ";;"
- (heredoc_start)
- ] @punctuation.delimiter
-
-[
- "$"
-] @punctuation.special
+] @punctuation.delimiter
```
Comment (amaanq) [inline queries/bash/highlights.scm:12]: add '&' as well

File: queries/bash/injections.scm [hunk 3]
```diff
@@ -1,11 +1,13 @@
...
- (heredoc_body) @injection.content
- (#offset! @injection.content 0 0 -1 0)
- (#downcase! @injection.language))(#set! injection.language "regex"))
+((heredoc_redirect
+  [
+    (heredoc_body)
+    (simple_heredoc_body)
+  ] @injection.content
+  (heredoc_end) @injection.language)
+ (#downcase! @injection.language))
```
Comment (amaanq) [inline queries/bash/injections.scm:13]: do we want to support injecting when ${THIS_SH} is detected?

https://git.savannah.gnu.org/tree/tests/arith-for.tests?id=ec8113b9861375e4e17b3307372569d429dec814#n91

File: queries/bash/highlights.scm (hunk 1 above)
Comment (ObserverOfTime) [reply queries/bash/highlights.scm:92]: I kept it the same as it was and this way it matches the closing bracket.

Comment (amaanq) [reply queries/bash/highlights.scm:92]: you can make the matching closing bracket in *this* case special, it'd be most useful for arith expansions imo

File: queries/bash/injections.scm (hunk 3 above)
Comment (ObserverOfTime) [reply queries/bash/injections.scm:13]: That link isn't working.

File: queries/bash/highlights.scm (hunk 1 above)
Comment (ObserverOfTime) [reply queries/bash/highlights.scm:92]: What about "(("?

Comment (amaanq) [reply queries/bash/highlights.scm:92]:
```scm
(arithmetic_expansion [ "$((" "((" ] @punctuation.special "))" @punctuation.special)
```

File: queries/bash/injections.scm (hunk 3 above)
Comment (amaanq) [reply queries/bash/injections.scm:13]: updated

Comment (ObserverOfTime) [reply queries/bash/injections.scm:13]: There are too many cases to cover.

```bash
${THIS_SH} -c '...'
sh -c '...'; /bin/sh -c '...'; /usr/bin/sh -c '...'
bash -c '...'; /bin/bash -c '...'; /usr/bin/bash -c '...'
command sh -c '...' # plus path variants
command bash -c '...' # plus path variants
command -p sh -c '...' # plus path variants
command -p bash -c '...' # plus path variants
env sh -c '...' # plus path variants, plus env options
env bash -c '...' # plus path variants, plus env options
```

Comment (amaanq) [reply queries/bash/injections.scm:13]: hm yeah, nevermind

**Review by amaanq — APPROVED** [review]

**Comment by amaanq** [comment]
sorta related, it'd be nice to match heredoc injections w/ the language ft/shorthand name too

e.g.
```bash
node <<JS
console.log("hi")
JS
```

not the end of the world

**Pull request merged**